lld/ELF/Writer.cpp

```diff
@@ -1102,6 +1110,44 @@ template <class ELFT> void Writer<ELFT>::setReservedSymbolSections() {
       }
     }
   }
+
+  // Setup nanoMIPS _gp if not set up by linker script yet
+  // TODO: Check the constraints that are marked in documentation, chapter 5.3
+  // Sections -> Special sections
+  // http://codescape.mips.com/components/toolchain/nanomips/2019.03-01/docs/MIPS_nanoMIPS_ABI_supplement_01_03_DN00179.pdf
+  // also, check if we can do something about the order of the small data area
+  // sections
+  // TODO: Needs testing and verification
+  if (ElfSym::nanoMipsGp && !ElfSym::nanoMipsGp->scriptDefined) {
+    const char *smallDataArea[] = {".got", ".ssdata", ".ssbss", ".sdata",
+                                   ".sbss"};
+    // Refers to the smallDataArea section that is found
+    int foundSec = 5;
+    size_t foundSecIdx = 0;
+    for (auto [i, osec] : llvm::enumerate(outputSections)) {
+      for (int j = 0; j < foundSec; j++) {
+        if (osec->name == smallDataArea[j]) {
+          foundSec = j;
+          foundSecIdx = i;
+        }
+        break;
```

Comment:
Is it intentional to break here without checking other names than ".got"?

Reply:
Yup that's an error, thanks for pointing that out. The break should be a part of the if clause.